reporting/spacewalk-report

```diff
@@ -190,11 +190,11 @@ if __name__ == '__main__':
             for column in where:
                 if column not in report.columns:
                     systemExit(-6, 'Unknown column [%s] in report [%s].' % (column, report_name))
-                for v in (val for _, vals in where[column].iteritems() for val in vals):
+                for v in (val for _, vals in list(where[column].items()) for val in vals):
                     if report.column_types[column] == 'i' and not re.match('^[0-9]+$', v):
                         systemExit(-7, 'Column [%s] in report [%s] only accepts integer value.' % (column, report_name))
 
-                for clause, values in where[column].iteritems():
+                for clause, values in list(where[column].items()):
```

Comment:
Why you're converting generator into the list (requires loop over it and `malloc`) and then iterate over it all over again?

Reply:
:+1:

Reply:
Actuall `dict.items()` is not a generator, but it returns different type depending on the Python version.

- Python 2 -> `list`
- Python 3 -> `dict_items`

In this particular case, we're only iterating over the items, so there is no need of casting to `list`.

Reply:
My point is, that since you already use `six`, you should then use it here as well:

```python
for clause, values in six.iteritems(where[column]):
   ...
```